### PR TITLE
nft, doc: Document public package, functions and methods

### PR DESCRIPTION
**Pull request by EdDev**

### CODE REVIEW

File: nft/chain.go [hunk 1]
```diff
@@ -50,9 +50,14 @@ const (
...
 )
 
+// NewRegularChain returns a new schema chain structure for a regular chain.
 func NewRegularChain(table *schema.Table, name string) *schema.Chain {
 	return NewChain(table, name, nil, nil, nil, nil)
 }
+
+// NewChain returns a new schema chain structure for a base chain.
+// For base chains, all arguments are required except the policy.
+// Note: Missing arguments will cause an error once the config is applied.
```
Comment (qinqon) [inline nft/chain.go]: Maybe we can link to nftables man with requirement ?

Also the "Note: " prefix is not needed there is similar explanations below and they don't have a "Note: " wich make sense.

File: nft/chain.go [hunk 2]
```diff
@@ -76,21 +81,36 @@ func NewChain(table *schema.Table, name string, ctype *ChainType, hook *ChainHoo
...
+// Adding multiple times the same chain has no affect when the config is applied.
 func (c *Config) AddChain(chain *schema.Chain) {
 	nftable := schema.Nftable{Chain: chain}
 	c.Nftables = append(c.Nftables, nftable)
 }
 
+// DeleteChain appends a given chain to the nftable config
+// with the `delete` action.
+// Attempting to delete a non-existing chain, results with a failure when the config is applied.
+// Note: The chain must not contain any rules or be used as a jump target.
```
Comment (qinqon) [inline nft/chain.go]: No need for "Note: " prefix

File: nft/config.go [hunk 3]
```diff
@@ -29,23 +29,31 @@ type Config struct {
...
 	if err := json.Unmarshal(data, c); err != nil {
 		return err
 	}
 	return nil
 }
 
+// FlushRuleset adds a command to the nftables config that erases all the configuration when applied.
+// It is commonly used as the first config instruction, followed by a declarative configuration.
+// When used, any previous configuration is flushed away before adding the new one.
+// Note that calling FlushRuleset has affect only on the configuration and will take affect only
```
Comment (qinqon) [inline nft/config.go]: Do we use "Note: " here ? or we remove it totally ?

**Review by qinqon — CHANGES_REQUESTED** [review]
It gives a lot of detail on functions, I think the "Note: " prefix is not needed

Does it make sense to put some "package" documentation under `doc.go` go file at nft package?

File: nft/chain.go (hunk 1 above)
Comment (EdDev) [reply nft/chain.go]: You mean to drop the `note: ` prefix or the whole line?

Comment (EdDev) [reply nft/chain.go]: > Maybe we can link to nftables man with requirement ?

It is mentioned in #20

File: nft/chain.go (hunk 2 above)
Comment (EdDev) [reply nft/chain.go]: Done

File: nft/chain.go (hunk 1 above)
Comment (EdDev) [reply nft/chain.go]: I will add it to the package level in `doc.go` as you suggested.

File: nft/config.go (hunk 3 above)
Comment (EdDev) [reply nft/config.go]: Removed, done.

**Review by EdDev — COMMENTED** [review]
> It gives a lot of detail on functions, I think the "Note: " prefix is not needed

I removed all "Note" wording.

> Does it make sense to put some "package" documentation under doc.go go file at nft package?

Done.

**Review by qinqon — APPROVED** [review]

**Pull request merged**